packages/cli/tests/unit/agentInstall/agentInstallerProcedure.spec.ts

```diff
@@ -0,0 +1,36 @@
+import AgentInstallerProcedure from '../../../src/cmds/agentInstaller/agentInstallerProcedure';
+import * as CommandRunner from '../../../src/cmds/agentInstaller/commandRunner';
+
+import fs from 'fs';
+import { TestAgentInstaller } from './TestAgentProcedure';
+import UI from '../../../src/cmds/userInteraction';
+
+jest.mock('../../../src/cmds/userInteraction');
+jest.mock('../../../src/cmds/agentInstaller/commandRunner');
+
+const { prompt, success, warn } = jest.mocked(UI);
+const { run } = jest.mocked(CommandRunner);
+
+const procedure = new AgentInstallerProcedure(new TestAgentInstaller(), '/test/path');
+
+describe(AgentInstallerProcedure, () => {
+  it('prints any warnings from the validator', async () => {
+    prompt.mockResolvedValue({ confirm: true });
+    jest.spyOn(procedure, 'validateProject').mockResolvedValue(
+      { errors: [{ level: 'warning', message: 'Remember to foo the bar.', help_urls: ['test:///help/url'] }] }
+    );
+    run.mockResolvedValue({ stdout: '{"configuration": {"contents": ""}}', stderr: '' });
+    jest.spyOn(fs, 'writeFileSync').mockImplementation();
+
+    await procedure.run();
+
+    expect(success).toBeCalled();
+
+    expect(warn).toBeCalled();
+    const [message] = warn.mock.calls[0];
+    expect(message).toContain('Warning: Remember to foo the bar.');
+    expect(message).toContain('test:///help/url');
+  });
+
+  beforeEach(jest.restoreAllMocks);
+});
```

Comment:
Is it worth also adding a test that produces an `error` instead of a `warning`?

Reply:
That's a good point; maybe not here but in the agentProcedure spec.